Mauve XMFA AlignIO Parser

erasche opened the pull request:
http://darlinglab.org/mauve/user-guide/files.html

peterjc commented:
The input example file should also be added to ``test_AlignIO.py`` to check reading from Mauve and writing to other formats works. Note this is a old style print-and-compare test, see the chapter in the tutorial for background - you will need ``python run_tests.py -g test_AlignIO.py`` to generate the revised expected output.

erasche commented:
@peterjc addressed concerns of pythonic counting, and adding to `test_AlignIO.py`. Now fighting through gracefully handling bad data, so I've marked this WIP and will ping you again when it passes all of its tests.

erasche commented:
@peterjc I believe I've finally got Mauve passing all of the tests!! (Except the last one which seems stuck? :/)

I've addressed all of your concerns mentioned above in-code. IDs are unique, pythonic counting is correct and under test. 

The code is not as clean and beautiful as it could be, but it's functional and has to work around some of the built-in limitations (e.g. stockholm's IDs vs needing unique IDs, etc,) so I'm happy with this as-is. Please let me know what, if any, concerns you have that I still need to address in code.

erasche commented:
(low-priority) ping @peterjc

erasche commented:
(low-priority) ping @peterjc

snayfach commented:
It doesn't look like Bio/AlignIO/MauveIO.py was ever merged into the main repository, right? If so, what's the issue? It would be super useful.

erasche commented:
@snayfach I have long since given up on getting this merged (e.g. "hey, ping, what's up? what's missing?" comments were deleted, which is fair, they are perhaps not constructive.) 

If you need to parse XMFA, I wrote a [simple XMFA](https://github.com/TAMU-CPT/galaxy-tools/blob/master/tools/comparative/xmfa.py) parser that my org uses instead of a proper BioPython one (which we'd still really, really love!) It doesn't generate BioPython compatible objects, but we needed *something* and didn't want to have to fork biopython just for my organisation. It's BSD licensed.

peterjc commented:
Unfortunately I'd forgotten about this, thank you for taking a look at it @snayfach - having never used XMFA myself, its harder to know how important it might be.

Looking back over the comments I may have wrongly assumed from @erasche's note about test working *"(Except the last one which seems stuck? :/)"* that it was still a work in progress.

From a quick skim over the changes I doubt the merge conflicts are very substantial, after all adding a new parser is largely self contained except for the general AlignIO tests.

@erasche: You've mentioned licensing a simpler XMFA parser under the BSD license. I presume given your comment on #898 you are also happy to dual licence this offered contribution under both the _Biopython License Agreement_ AND the _BSD 3-Clause License_?

erasche commented:
Yes, you may consider this code dual licensed.

Peter Cock <notifications@github.com> schrieb am Di., 14. März 2017, 23:30:

Unfortunately I'd forgotten about this, thank you for taking a look at it
@snayfach <https://github.com/snayfach> - having never used XMFA myself,
its harder to know how important it might be.

Looking back over the comments I may have wrongly assumed from @erasche
<https://github.com/erasche>'s note about test working *"(Except the last
one which seems stuck? :/)"* that it was still a work in progress.

From a quick skim over the changes I doubt the merge conflicts are very
substantial, after all adding a new parser is largely self contained except
for the general AlignIO tests.

@erasche <https://github.com/erasche>: You've mentioned licensing a simpler
XMFA parser under the BSD license. I presume given your comment on #898
<https://github.com/biopython/biopython/issues/898> you are also happy to
dual licence this offered contribution under both the *Biopython License
Agreement* AND the *BSD 3-Clause License*?

—
You are receiving this because you were mentioned.


Reply to this email directly, view it on GitHub
<https://github.com/biopython/biopython/pull/544#issuecomment-286594426>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AAb_u4cnVkgXiUXTRj0QtnDw5reg0Hywks5rlyMIgaJpZM4Ea8DZ>
.

peterjc commented:
Thanks @erasche.

peterjc commented:
Rebased version here, including proposed ``NEWS.rst`` entry: https://github.com/peterjc/biopython/tree/mauveio

TravisCI passed: https://travis-ci.org/peterjc/biopython/builds/219753774

Does that look OK to you Eric?

Looking at this again reminded me we've not really solved what to do with record identifiers/start-end in any consistent way for Biopython - the can of worms I alluded to earlier - but trust these change's won't block any future changes in that area.

erasche commented:
Yep, looks good to me @peterjc. Thank you for rebasing this!
